catkin_ws/src/controls/src/servers/state_quaternion_pid_server.py

```diff
@@ -14,36 +14,40 @@ def __init__(self):
         self.server = actionlib.SimpleActionServer('state_quaternion_server', StateQuaternionAction, execute_cb=self.callback, auto_start=False)
         self.previous_goal_quat = None
         # Calculation parameters/values
-        
+        self.goal_position = None
+        self.goal_quat = None
         self.server.start()        
 
     def callback(self, goal):
         print("\n\nQuaternion Server got goal:\n",goal)
         self.cancelled = False
-        self.goal = goal
         if self.pose is not None:
-            if(self.goal.displace.data):
-                goal_position, goal_quat = self.get_goal_after_displace()
+            if(goal.displace.data):
+                goal_position, goal_quat = self.get_goal_after_displace(goal)
+                self.goal_position = goal_position
+                self.goal_quat = goal_quat
             else:
-                goal_position = [self.goal.pose.position.x, self.goal.pose.position.y, self.goal.pose.position.z]
-                goal_quat = np.quaternion(self.goal.pose.orientation.w, self.goal.pose.orientation.x, self.goal.pose.orientation.y, self.goal.pose.orientation.z)
+                goal_position = [goal.pose.position.x, goal.pose.position.y, goal.pose.position.z]
```

Comment:
The [StateQuaternionAction](https://github.com/mcgill-robotics/AUV-2023/blob/noetic/catkin_ws/src/auv_msgs/action/StateQuaternion.action) definition does contain a Pose - am I missing something? Later on you pass what I assume are meant to be booleans to the check_settled function. It seems you have a mix up of message data